components/HomeScreenButtons.js

```diff
@@ -0,0 +1,193 @@
+  const login = () => {
+    dispatch({ type: SHOW_LOGIN_FORM });
+  };
+  return (
+    <TouchableHighlight style={styles.authButton} onPress={login}>
+      <Text id="login-button" style={styles.buttonText}>
+        Login
+      </Text>
+    </TouchableHighlight>
+  );
+};
+
+const SignUpButton = () => {
+  const dispatch = useDispatch();
+  const navigation = useNavigation();
+  return (
+    <TouchableHighlight
+      style={styles.authButton}
+      onPress={() => {
+        navigation.navigate("Sign up", {
+          name: "Sign up",
+        });
+        dispatch({
+          type: SIGN_UP,
+          payload: { communityId: undefined },
+        });
+      }}
+    >
+      <Text id="sign-up-button" style={styles.buttonText}>
+        Sign up
+      </Text>
+    </TouchableHighlight>
+  );
+};
+
+const TripPingsButton = () => {
+  const dispatch = useDispatch();
+  const navigation = useNavigation();
+  return (
+    <TouchableHighlight
+      style={styles.button}
+      onPress={() => {
+        fetchTrips(dispatch);
+        navigation.navigate("Trips", {
+          name: "Trips",
+        });
+      }}
+    >
+      <Text id="trip-list-button" style={styles.buttonText}>
+        Trip Pings
+      </Text>
+    </TouchableHighlight>
+  );
+};
+
+const PingBoardButton = () => {
+  const navigation = useNavigation();
+  return (
+    <TouchableHighlight
+      style={styles.button}
+      onPress={() => {
+        navigation.navigate("My Ping Board", {
+          name: "My Ping Board",
+        });
+      }}
+    >
+      <Text id="request-list-button" style={styles.buttonText}>
+        My Ping Board
+      </Text>
+    </TouchableHighlight>
+  );
+};
+
+const PongBoardButton = () => {
+  const dispatch = useDispatch();
+  const userId = useSelector((state) => state.userId);
+  const navigation = useNavigation();
+  return (
+    <TouchableHighlight
+      style={styles.button}
+      onPress={() => {
+        navigation.navigate("My Pong Board", {
+          name: "My Pong Board",
+        });
+        getRequestInformation(userId, dispatch);
+      }}
+    >
+      <Text id="request-button" style={styles.buttonText}>
+        My Pong Board
+      </Text>
+    </TouchableHighlight>
+  );
+};
+
+const ProfileButton = () => {
+  const dispatch = useDispatch();
+  const userId = useSelector((state) => state.userId);
+  const navigation = useNavigation();
+  return (
+    <TouchableHighlight
+            style={styles.button}
+            onPress={() => {
+              navigation.navigate("My Profile", {
+                name: "My Profile",
+              });
+              dispatch({
+                type: AUTHENTICATE,
+                payload: { updateProfileMessage: "" },
+              });
+              getProfileInformation(userId, dispatch);
+            }}
+          >
+            <Text id="profile-button" style={styles.buttonText}>
+              My Profile
+            </Text>
+          </TouchableHighlight>
+  );
```

Comment:
Im not sure if the formatting is off here :)

Reply:
Reviewed and corrected :)